store/mockstore/mocktikv/mvcc_leveldb.go

```diff
@@ -931,6 +941,102 @@ func (mvcc *MVCCLevelDB) Cleanup(key []byte, startTS uint64) error {
 	return mvcc.db.Write(batch, nil)
 }
 
+// CheckTxnStatus checks the primary lock of a transaction to decide its status.
+// The return values are (ttl, commitTS, err):
+// If the transaction is active, this function returns the ttl of the lock;
+// If the transaction is committed, this function returns the commitTS;
+// If the transaction is rollbacked, this function returns (0, 0, nil)
+// Note that CheckTxnStatus may also push forward the `minCommitTS` of the
+// transaction, so it's not simply a read-only operation.
+//
+// primaryKey + lockTS together could locate the primary lock.
+// callerStartTS is the start ts of reader transaction.
+// currentTS is the current ts, but it may be inaccurate. Just use it to check TTL.
+func (mvcc *MVCCLevelDB) CheckTxnStatus(primaryKey []byte, lockTS, callerStartTS, currentTS uint64) (uint64, uint64, error) {
+	mvcc.mu.Lock()
+	defer mvcc.mu.Unlock()
+
+	startKey := mvccEncode(primaryKey, lockVer)
+	iter := newIterator(mvcc.db, &util.Range{
+		Start: startKey,
+	})
+	defer iter.Release()
+
+	if iter.Valid() {
+		dec := lockDecoder{
+			expectKey: primaryKey,
+		}
+		ok, err := dec.Decode(iter)
+		if err != nil {
+			return 0, 0, errors.Trace(err)
+		}
+		// If current transaction's lock exists.
+		if ok && dec.lock.startTS == lockTS {
+			lock := dec.lock
+			batch := &leveldb.Batch{}
+
+			// If the lock has already outdated, clean up it.
+			if uint64(oracle.ExtractPhysical(lock.startTS))+lock.ttl < uint64(oracle.ExtractPhysical(currentTS)) {
+				if err = rollbackLock(batch, lock, primaryKey, lockTS); err != nil {
+					return 0, 0, errors.Trace(err)
+				}
+				if err = mvcc.db.Write(batch, nil); err != nil {
+					return 0, 0, errors.Trace(err)
+				}
+				return 0, 0, nil
+			}
+
+			// If this is a large transaction and the lock is active, push forward the minCommitTS.
+			// lock.minCommitTS == 0 may be a secondary lock, or not a large transaction.
+			if lock.minCommitTS > 0 {
+				// We *must* guarantee the invariance lock.minCommitTS >= callerStartTS + 1
+				if lock.minCommitTS < callerStartTS+1 {
+					lock.minCommitTS = callerStartTS + 1
+
+					// Remove this condition should not affect correctness.
+					// We do it because pushing forward minCommitTS as far as possible could avoid
+					// the lock been pushed again several times, and thus reduce write operations.
+					if lock.minCommitTS < currentTS {
+						lock.minCommitTS = currentTS
+					}
+
+					writeKey := mvccEncode(primaryKey, lockVer)
+					writeValue, err := lock.MarshalBinary()
```

Comment:
Should add `minCommitTS` in function `MarshalBinary ` and the `UnmarshalBinary` function?

Reply:
Done